util/dataset.go

```diff
@@ -147,17 +150,61 @@ func (d *DataSet) Validate() error {
 	}
 
 	if d.IngestPipeline == "" && len(paths) > 0 {
-		return fmt.Errorf("Package contains pipelines which are not used: %v, %s", paths, d.ID)
+		return fmt.Errorf("unused pipelines in the package (dataSetID: %s): %s", d.ID, strings.Join(paths, ","))
 	}
 
 	// In case an ingest pipeline is set, check if it is around
 	if d.IngestPipeline != "" {
-		_, errJSON := os.Stat(filepath.Join(pipelineDir, d.IngestPipeline+".json"))
-		_, errYAML := os.Stat(filepath.Join(pipelineDir, d.IngestPipeline+".yml"))
+		var validFound bool
+
+		jsonPipelinePath := filepath.Join(pipelineDir, d.IngestPipeline+".json")
+		_, errJSON := os.Stat(jsonPipelinePath)
+		if errJSON != nil && !os.IsNotExist(errJSON) {
+			return errors.Wrapf(errJSON, "stat ingest pipeline JSON file failed (path: %s)", jsonPipelinePath)
+		} else if !os.IsNotExist(errJSON) {
```

Comment:
I don't think you need an else here as if it matches before, it will return.

Reply:
Fixed